### ISSUE TITLE
@TargetLocations on a type qualifier has no effect.

### ISSUE THREAD

**Issue by t-rasmud** [typetools/checker-framework#3495]
The type qualifier `@LowerBoundBottom` in the `index` checker is annotated with the meta annotation `@TargetLocations({TypeUseLocation.EXPLICIT_LOWER_BOUND, TypeUseLocation.EXPLICIT_UPPER_BOUND})`.
This implies that `@LowerBoundAnnotation` should not be written by programmers in locations other than explicit bounds.

Consider the following test case:
```
import org.checkerframework.checker.index.qual.LowerBoundBottom;

public class InvalidBottom {
    void test(@LowerBoundBottom int x) {}
}
```

Running the index checker on this example with the command `javac -processor index checker/tests/index/InvalidBottom.java` doesn't report any warning. This is incorrect because `@LowerBoundBottom` should not be allowed to be written explicitly on a parameter type.

**Comment by wmdietl** [comment]
There already is an issue for this: #1919. Can you work on fixing this?
